### PR TITLE
Refactor Slice.astep

### PR DESCRIPTION
**Pull request by ColCarroll**
Cleaned up the imports, renamed some variables.  The only functional difference is changing the loop under `if self.tune` to be more efficient.  On my machine, this change lowered the time for `test_step` from ~80s to ~55s.

### CODE REVIEW

**Review by fonnesbeck — APPROVED** [review]
Good to merge once tests pass.

**Comment by ColCarroll** [comment]
Thanks -- will rebase once #1372 is merged!

**Comment by twiecki** [comment]
Thanks!

**Pull request merged**
